gensim/sklearn_api/d2vmodel.py

```diff
@@ -63,8 +64,9 @@ def fit(self, X, y=None):
         Fit the model according to the given training data.
         Calls gensim.models.Doc2Vec
         """
+        d2v_sentences = [doc2vec.TaggedDocument(words[0], [i]) for i, words in enumerate(X)]
```

Comment:
1. Can you check X, this can be already in `TaggedDocument` format (no need to convert it directly)
2. Why `words[0]`? If `X` is iterable of the list of tokens, `words[0]` will be one token only.
3. Please add tests for `fit_transform`